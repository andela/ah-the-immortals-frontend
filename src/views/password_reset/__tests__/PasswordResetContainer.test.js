import React from 'react';
import { Router } from 'react-router-dom';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { createBrowserHistory } from 'history';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import toJson from 'enzyme-to-json';
import { Form } from 'react-bootstrap';
import PasswordResetForm from '../PasswordResetContainer';

describe('Tests PasswordResetContainer', () => {
  const store = configureStore([thunk])({
    resetpasswordlink: { message: {}, errors: {} }
  });
  const history = createBrowserHistory();
  const props = {
    showModal: jest.fn(),
  };
  const wrapper = mount(
    <Router history={history}>
      <Provider store={store}>
        <PasswordResetForm {...props} />
      </Provider>
    </Router>
  );
  it('Tests mounting of component', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('Tests input change in form', () => {
    const resetBtn = wrapper.findWhere(n => n.type() === 'button' && n.contains('Find account'));
    resetBtn.simulate('click', {
      preventDefault: jest.fn()
    });
    const form = wrapper.find(Form);
    const input = form.find('FormControl[name="email"]');
    input.simulate('change', {
      preventDefault: jest.fn(),
      target: {
        value: 'issamwangi@gmail.com',
        name: 'email'
      }
    });
    expect(input.length).toEqual(1);
  });
  it('handles submit event', () => {
    const resetBtn = wrapper.findWhere(n => n.type() === 'button' && n.contains('Find account'));
    resetBtn.simulate('submit', {
      preventDefault: jest.fn()
    });
    expect(resetBtn.length).toEqual(1);
  });
});
