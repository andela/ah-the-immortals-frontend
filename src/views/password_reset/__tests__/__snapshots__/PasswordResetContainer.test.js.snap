// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests PasswordResetContainer Tests mounting of component 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Provider
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <Connect(PasswordResetForm)
      showModal={[MockFunction]}
    >
      <PasswordResetForm
        errors={Object {}}
        message={Object {}}
        resetPasswordLinkAction={[Function]}
        showModal={[MockFunction]}
      >
        <div>
          <NavigationBar>
            <div>
              <Bootstrap(undefined)
                className="homenav fixed-top"
              >
                <ForwardRef
                  bsPrefix="navbar"
                  className="homenav fixed-top"
                  innerRef={null}
                >
                  <Uncontrolled(Navbar)
                    bsPrefix="navbar"
                    className="homenav fixed-top"
                    innerRef={null}
                  >
                    <Navbar
                      bsPrefix="navbar"
                      className="homenav fixed-top"
                      collapseOnSelect={false}
                      expand={true}
                      onToggle={[Function]}
                      variant="light"
                    >
                      <nav
                        className="homenav fixed-top navbar navbar-expand navbar-light"
                      >
                        <NavbarBrand>
                          <span
                            className="navbar-brand"
                          >
                            <Link
                              className="brand-text"
                              to="/"
                            >
                              <a
                                className="brand-text"
                                href="/"
                                onClick={[Function]}
                              >
                                <b>
                                  Authors Haven
                                </b>
                              </a>
                            </Link>
                          </span>
                        </NavbarBrand>
                        <NavbarCollapse>
                          <Collapse
                            appear={false}
                            dimension="height"
                            getDimensionValue={[Function]}
                            in={false}
                            mountOnEnter={false}
                            timeout={300}
                            unmountOnExit={false}
                          >
                            <Transition
                              addEndListener={[Function]}
                              appear={false}
                              aria-expanded={null}
                              enter={true}
                              exit={true}
                              in={false}
                              mountOnEnter={false}
                              onEnter={[Function]}
                              onEntered={[Function]}
                              onEntering={[Function]}
                              onExit={[Function]}
                              onExited={[Function]}
                              onExiting={[Function]}
                              timeout={300}
                              unmountOnExit={false}
                            >
                              <div
                                aria-expanded={null}
                                className="navbar-collapse collapse"
                              >
                                <Nav
                                  className="justify-content-end"
                                  fill={false}
                                  justify={false}
                                  style={
                                    Object {
                                      "width": "100%",
                                    }
                                  }
                                >
                                  <ContextTransform(AbstractNav)
                                    as="div"
                                    className="justify-content-end navbar-nav"
                                    onSelect={[Function]}
                                    style={
                                      Object {
                                        "width": "100%",
                                      }
                                    }
                                  >
                                    <AbstractNav
                                      as="div"
                                      className="justify-content-end navbar-nav"
                                      onSelect={[Function]}
                                      parentOnSelect={[Function]}
                                      style={
                                        Object {
                                          "width": "100%",
                                        }
                                      }
                                    >
                                      <div
                                        className="justify-content-end navbar-nav"
                                        onKeyDown={[Function]}
                                        style={
                                          Object {
                                            "width": "100%",
                                          }
                                        }
                                      >
                                        <NavItem>
                                          <div
                                            className="nav-item"
                                          >
                                            <Link
                                              to="#null"
                                            >
                                              <a
                                                href="/#null"
                                                onClick={[Function]}
                                              >
                                                <Button
                                                  active={false}
                                                  className="buttons"
                                                  disabled={false}
                                                  type="button"
                                                  variant="danger"
                                                >
                                                  <button
                                                    className="buttons btn btn-danger"
                                                    disabled={false}
                                                    type="button"
                                                  >
                                                    Join now
                                                  </button>
                                                </Button>
                                              </a>
                                            </Link>
                                          </div>
                                        </NavItem>
                                        <NavItem>
                                          <div
                                            className="nav-item"
                                          >
                                            <Link
                                              to="#null"
                                            >
                                              <a
                                                href="/#null"
                                                onClick={[Function]}
                                              >
                                                <Button
                                                  active={false}
                                                  className="buttons"
                                                  disabled={false}
                                                  type="button"
                                                  variant="outline-light"
                                                >
                                                  <button
                                                    className="buttons btn btn-outline-light"
                                                    disabled={false}
                                                    type="button"
                                                  >
                                                    Sign In
                                                  </button>
                                                </Button>
                                              </a>
                                            </Link>
                                          </div>
                                        </NavItem>
                                      </div>
                                    </AbstractNav>
                                  </ContextTransform(AbstractNav)>
                                </Nav>
                              </div>
                            </Transition>
                          </Collapse>
                        </NavbarCollapse>
                      </nav>
                    </Navbar>
                  </Uncontrolled(Navbar)>
                </ForwardRef>
              </Bootstrap(undefined)>
            </div>
          </NavigationBar>
          <Container
            className="reset"
            fluid={false}
          >
            <div
              className="reset container"
            >
              <div>
                <h1
                  className="resetmessage"
                >
                  First, let's find your account
                </h1>
              </div>
              <Form
                inline={false}
                onSubmit={[Function]}
              >
                <form
                  className=""
                  onSubmit={[Function]}
                >
                  <FormControl
                    className="confirm-input"
                    name="email"
                    onChange={[Function]}
                    placeholder="Please enter your email"
                    required={true}
                    type="email"
                  >
                    <input
                      className="confirm-input form-control"
                      name="email"
                      onChange={[Function]}
                      placeholder="Please enter your email"
                      required={true}
                      type="email"
                    />
                  </FormControl>
                  <div
                    className="reset-buttons"
                  >
                    <Link
                      to="/"
                    >
                      <a
                        href="/"
                        onClick={[Function]}
                      >
                        <Button
                          active={false}
                          className="reset-buttons"
                          disabled={false}
                          type="button"
                          variant="outline-danger"
                        >
                          <button
                            className="reset-buttons btn btn-outline-danger"
                            disabled={false}
                            type="button"
                          >
                            Cancel
                          </button>
                        </Button>
                      </a>
                    </Link>
                    <Button
                      active={false}
                      className="reset-buttons"
                      disabled={false}
                      type="submit"
                      variant="primary"
                    >
                      <button
                        className="reset-buttons btn btn-primary"
                        disabled={false}
                        type="submit"
                      >
                        Find account
                      </button>
                    </Button>
                  </div>
                </form>
              </Form>
            </div>
          </Container>
        </div>
      </PasswordResetForm>
    </Connect(PasswordResetForm)>
  </Provider>
</Router>
`;
